packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-swift-visitor.test.ts.snap

```diff
@@ -0,0 +1,264 @@
+      .field(listContainer.requiredList, is: .required, ofType: .embeddedCollection(of: String.self)),
+      .field(listContainer.requiredListOfRequired, is: .required, ofType: .embeddedCollection(of: StatusEnum.self)),
+      .field(listContainer.listOfRequired, is: .optional, ofType: .embeddedCollection(of: Bool.self)),
+      .field(listContainer.requiredListOfRequiredDates, is: .required, ofType: .embeddedCollection(of: Temporal.Date.self)),
+      .field(listContainer.listOfRequiredFloats, is: .optional, ofType: .embeddedCollection(of: Double.self)),
+      .field(listContainer.requiredListOfCustomTypes, is: .required, ofType: .embeddedCollection(of: CustomType.self)),
+      .field(listContainer.createdAt, is: .optional, isReadOnly: true, ofType: .dateTime),
+      .field(listContainer.updatedAt, is: .optional, isReadOnly: true, ofType: .dateTime)
+    )
+    }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should handle nullability of lists appropriately 3`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct CustomType: Embeddable {
+  var name: String?
+  var list: [Int?]?
+  var requiredList: [String?]
+  var requiredListOfRequired: [StatusEnum]
+  var listOfRequired: [Bool]?
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should render lists with HAS_MANY connection according to the spec 1`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct Todo: Model {
+  public let id: String
+  public var title: String
+  public var tasks: List<task?>
+  public var createdAt: Temporal.DateTime?
+  public var updatedAt: Temporal.DateTime?
+  
+  public init(id: String = UUID().uuidString,
+      title: String,
+      tasks: List<task?> = []) {
+    self.init(id: id,
+      title: title,
+      tasks: tasks,
+      createdAt: nil,
+      updatedAt: nil)
+  }
+  internal init(id: String = UUID().uuidString,
+      title: String,
+      tasks: List<task?> = [],
+      createdAt: Temporal.DateTime? = nil,
+      updatedAt: Temporal.DateTime? = nil) {
+      self.id = id
+      self.title = title
+      self.tasks = tasks
+      self.createdAt = createdAt
+      self.updatedAt = updatedAt
+  }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Should render lists with HAS_MANY connection according to the spec 2`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+extension Todo {
+  // MARK: - CodingKeys 
+   public enum CodingKeys: String, ModelKey {
+    case id
+    case title
+    case tasks
+    case createdAt
+    case updatedAt
+  }
+  
+  public static let keys = CodingKeys.self
+  //  MARK: - ModelSchema 
+  
+  public static let schema = defineSchema { model in
+    let todo = Todo.keys
+    
+    model.pluralName = \\"Todos\\"
+    
+    model.fields(
+      .id(),
+      .field(todo.title, is: .required, ofType: .string),
+      .hasMany(todo.tasks, is: .required, ofType: task.self, associatedWith: task.keys.todo),
```

Comment:
This will be `.required`. Hope this is expected @lawmicha

Reply:
isn't this PR only applicable to embedded types? why are we changing connection types?

Reply:
what is the previous `is:` field? `.optional` or `.required`?